Content.Server/_DV/StationEvents/Events/MinorMassMindSwapRule.cs

```diff
@@ -0,0 +1,102 @@
+
+/// <summary>
+/// Forces a mind swap on a small amount of non-insulated psionic entities.
+/// </summary>
+internal sealed class MinorMassMindSwapRule : StationEventSystem<MinorMassMindSwapRuleComponent>
+{
+    [Dependency] private readonly AudioSystem _audio = default!;
+    [Dependency] private readonly MindSwapPowerSystem _mindSwap = default!;
+    [Dependency] private readonly ChatSystem _chat = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly MobStateSystem _mobstateSystem = default!;
+
+    private TimeSpan _warningSoundLength;
+    private ResolvedSoundSpecifier _resolvedWarningSound = string.Empty;
+
+    protected override void Started(EntityUid uid, MinorMassMindSwapRuleComponent component, GameRuleComponent gameRule, GameRuleStartedEvent args)
+    {
+        base.Started(uid, component, gameRule, args);
+
+        _resolvedWarningSound = _audio.ResolveSound(component.SwapWarningSound);
+        _warningSoundLength = _audio.GetAudioLength(_resolvedWarningSound);
+
+        component.SwapTime = Timing.CurTime + component.Delay;
+        component.SoundTime = component.SwapTime - _warningSoundLength;
+
+        var announcement = Loc.GetString("minor-mass-mind-swap-event-announcement", ("time", component.Delay.TotalSeconds));
+        var sender = Loc.GetString("minor-mass-mind-swap-event-sender");
+
+        _chat.DispatchGlobalAnnouncement(announcement, sender, true, component.AnnouncementSound, Color.White);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<MinorMassMindSwapRuleComponent, GameRuleComponent>();
+        while (query.MoveNext(out var uid, out var comp, out var ruleComp))
+        {
+            if (comp.SoundTime != null && comp.SoundTime <= Timing.CurTime)
+            {
+                _audio.PlayGlobal(_resolvedWarningSound, Filter.Broadcast(), true);
+                comp.SoundTime = null;
+            }
+
+            if (comp.SwapTime == null || comp.SwapTime > Timing.CurTime)
+                continue;
+
+            SwapMinds(comp);
+            comp.SwapTime = null;
+            GameTicker.EndGameRule(uid, ruleComp);
+        }
+    }
+
+    private void SwapMinds(MinorMassMindSwapRuleComponent component)
+    {
+        List<EntityUid> psionicActors = new();
+
+        var query = EntityQueryEnumerator<PotentialPsionicComponent, MobStateComponent>();
+        while (query.MoveNext(out var psion, out _, out _))
+        {
+            if (_mobstateSystem.IsAlive(psion) && !HasComp<PsionicInsulationComponent>(psion)
+                && HasComp<ActorComponent>(psion))
+            {
+                // Only a list of Players
+                psionicActors.Add(psion);
+            }
+        }
+
+        // We go with 4 pairs for now
+        List<EntityUid> actorsToSwap = new();
+        var swapPairCount = _random.Next(1, 3);
```

Comment:
Add number of pairs to the rule component and set 3 as the default. Also add a bit of logic that if its 0 or less, it sets it to 1 (to prevent people being idiots in the code)